### PR TITLE
Adds BOLT-LMM support

### PR DESCRIPTION
**Pull request by evagradovich**
## Overview

This PR:

- Adds support for BOLT-LMM for running GWAS

## Changes

- Adds `containers/bolt_lmm` Dockerfile
- Adds `phenotype_colname` parameter to covariate + PCs concatenation - ensures that null phenotype values are removed and null values in covariates are converted to -9 
- Adds `bin/sim_data_gwas.py` script used for generating input data from simulated output of `hapgen_simulate` pipeline. Data used is in Results folder from https://cloudos.lifebit.ai/app/jobs/61d6bb268c574a01e8d53c50
- Created `conf/test_binary_bolt_lmm.config` for testing BOLT-LMM and added to CI tests
- Added processes to convert input vcf to bgen (necessary for BOLT-LMM) and run BOLT-LMM

## Test

Lifebit Cloud (Lifebit Internal Production workspace): https://cloudos.lifebit.ai/app/jobs/61d6f4738c574a01e8d548da

## Future improvements

- Plug in BOLT-LMM output into a report: https://github.com/lifebit-ai/gwas/issues/129
- Add BOLT-LMM to documentation: https://github.com/lifebit-ai/gwas/issues/128

### CODE REVIEW

File: containers/bolt_lmm/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+
+RUN apt-get update --allow-releaseinfo-change && \
+    apt-get install -y \
+    procps && \
+    apt-get clean -y 
+
+RUN wget https://storage.googleapis.com/broad-alkesgroup-public/BOLT-LMM/downloads/BOLT-LMM_v2.3.6.tar.gz
+
+RUN tar -xvf BOLT-LMM_v2.3.6.tar.gz
```
Comment (sk-sahu) [inline containers/bolt_lmm/Dockerfile:15]: A small suggestion to make the container light - 

```suggestion
RUN wget https://storage.googleapis.com/broad-alkesgroup-public/BOLT-LMM/downloads/BOLT-LMM_v2.3.6.tar.gz \
  && tar -xvf BOLT-LMM_v2.3.6.tar.gz \
  && rm BOLT-LMM_v2.3.6.tar.gz
```

File: main.nf [hunk 2]
```diff
@@ -413,10 +430,11 @@ process run_pca {
     input:
     set val(plink_prefix), file(bed), file(bim), file(fam) from ch_plink_pruned_for_pca
     each file(phenotype_file) from ch_pheno_pca
+    file("concat_covariates.R") from ch_concat_covariates_r
```
Comment (sk-sahu) [inline main.nf:433]: I think having a variable name in input will be more appropriate in-case they use in script block. For now, in the script block, it is just able to read the `concat_covariates.R` as it is already able to copy to work directory.

```suggestion
    file(concat_covariates_script) from ch_concat_covariates_r
```

File: main.nf [hunk 3]
```diff
@@ -438,6 +456,7 @@ process run_pca {
     fi
 
     concat_covariates.R \
+    --phenotype_colname=${params.phenotype_colname} \
```
Comment (sk-sahu) [inline main.nf:459]: I see this `pararms.phenotype_colname` is a required parameter, do you want to write an exit logic in-case this is not provided!!

There is a default value, if this is sufficient then the above statement not required. 

https://github.com/lifebit-ai/gwas/blob/acbb091696d1c8da3b538144f8a6d6ef873585d8/nextflow.config#L60

**Review by sk-sahu — COMMENTED** [review]
LGTM 👍 just some minor comments

File: main.nf (hunk 3 above)
Comment (evagradovich) [reply main.nf:463]: Thank you @sk-sahu - exit logic added in https://github.com/lifebit-ai/gwas/pull/127/commits/3f1fd9ddeff655f5ec0573b2a5af2ab1064fc92b

**Review by sk-sahu — APPROVED** [review]
Thanks for adding the suggestion 

Now LGTM 👍

**Pull request merged**